101-azurefirewall-with-zones-sandbox/azuredeploy.json

```diff
@@ -0,0 +1,397 @@
+    "serverNicName": "ServerNic",
+    "serverSubnetId": "[resourceId('Microsoft.Network/virtualNetworks/subnets', parameters('virtualNetworkName'), variables('serversSubnetName'))]",
+    "storageAccountName": "[concat(uniquestring(resourceGroup().id), 'sajumpbox')]",
+    "azfwRouteTableName": "AzfwRouteTable",
+    "firewallName": "firewall1",
+    "publicIPName": "publicIP1",
+    "azureFirewallSubnetId":  "[resourceId('Microsoft.Network/virtualNetworks/subnets',parameters('virtualNetworkName'), variables('azureFirewallSubnetName'))]",
+    "azureFirewallPublicIpId": "[resourceId('Microsoft.Network/publicIPAddresses',variables('publicIPName'))]"
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Storage/storageAccounts",
+      "name": "[variables('storageAccountName')]",
+      "apiVersion": "2017-10-01",
+      "location": "[parameters('location')]",
+      "sku": {
+        "name": "Standard_LRS"
+      },
+      "kind": "Storage",
+      "properties": {}
+    },
+    {
+      "type": "Microsoft.Network/routeTables",
+      "name": "[variables('azfwRouteTableName')]",
+      "apiVersion": "2018-02-01",
+      "location": "[parameters('location')]",
+      "properties": {
+        "disableBgpRoutePropagation": false,
+        "routes": [
+          {
+            "name": "AzfwDefaultRoute",
+            "properties": {
+              "addressPrefix": "0.0.0.0/0",
+              "nextHopType": "VirtualAppliance",
+              "nextHopIpAddress": "[variables('nextHopIP')]"
+            }
+          }
+        ]
+      }
+    },
+    {
+      "name": "[parameters('virtualNetworkName')]",
+      "apiVersion": "2017-06-01",
+      "type": "Microsoft.Network/virtualNetworks",
+      "location": "[parameters('location')]",
+      "dependsOn": [
+        "[resourceId('Microsoft.Network/routeTables', variables('azfwRouteTableName'))]"
+      ],
+      "tags": {
+        "displayName": "[parameters('virtualNetworkName')]"
+      },
+      "properties": {
+        "addressSpace": {
+          "addressPrefixes": [
+            "[variables('vnetAddressPrefix')]"
+          ]
+        },
+        "subnets": [
+          {
+            "name": "[variables('jumpBoxSubnetName')]",
+            "properties": {
+              "addressPrefix": "[variables('jumpboxSubnetPrefix')]"
+            }
+          },
+          {
+            "name": "[variables('azureFirewallSubnetName')]",
+            "properties": {
+              "addressPrefix": "[variables('azureFirewallSubnetPrefix')]"
+            }
+          },
+          {
+            "name": "[variables('serversSubnetName')]",
+            "properties": {
+              "addressPrefix": "[variables('serversSubnetPrefix')]",
+              "routeTable": {
+                "id": "[resourceId('Microsoft.Network/routeTables', variables('azfwRouteTableName'))]"
+              }
+            }
+          }
+        ]
+      }
+    },
+    {
+      "name": "[variables('publicIPName')]",
```

Comment:
since we're deploying 2 of everything, put these into a copy loop

Reply:
We have it this way in 101-azurefirewall-sandbox/azuredeploy.json so took it from there. Still change it here?

Reply:
yes, we should actually look at combining these samples if they are so similar

Reply:
Idea was to have separate templates with availability zones to make it more explicit. But I could choose to remove old ones as this new one is a superset with zones being there by default now. Should I do that?

Reply:
yeah - it looks like the only addition in these two is the ability to use zones - so we should replace the ones without zones, with these....

Reply:
Ok I will remove the existing files then. Also I am not too familiar with how the copy loop works so will have to understand that before I could make that change.

Reply:
Actually if I use a copyloop, the name for each of these would be more generic right? I believe we had it this way to make it more clear on the jump box vs server machines and similar things. Do you have any suggestions for that?

Reply:
No you can use whatever names you want - for example you can create an array of names and reference the array item with copyIndex,e.g.

variables:{
"vmNames": [ "foo", "bar" ] …

Reply:
While I can use that variable array for names but it goes not buy much and the things which we create 2 of, they are not identical and will just make the template more complicated